feat(react-scheduler): add capability to edit recurrent appointments to appointment form (#2205)
DevExpress/devextreme-reactive: packages/dx-react-scheduler/docs/reference/appointment-form.md

AppointmentForm.CommandLayout | [AppointmentForm.CommandLayoutProps](#appointmentformcommandlayoutprops) | A component that renders a layout for command buttons.
AppointmentForm.BasicLayout | [AppointmentForm.BasicLayoutProps](#appointmentformbasiclayoutprops) | A component that renders a layout for editors that edit basic appoinement data.
AppointmentForm.RecurrenceLayout | [AppointmentForm.RecurrenceLayoutProps](#appointmentformrecurrencelayoutprops) | A component that renders a layout for editors that specify the appointment's recurrence.
AppointmentForm.TextEditor | [AppointmentForm.TextEditorProps](#appointmentformtexteditorprops) | A component that renders a text editor.
AppointmentForm.DateEditor | [AppointmentForm.DateEditorProps](#appointmentformdateeditorprops) | A component that renders a date-time editor.
AppointmentForm.Label | [AppointmentForm.LabelProps](#appointmentformlabelprops) | A component that renders a text label.
AppointmentForm.BooleanEditor | [AppointmentForm.BooleanEditorProps](#appointmentformbooleaneditorprops) | A component that renders an editor of Boolean values.

> AppointmentForm.Select | [AppointmentForm.SelectProps](#appointmentformselectprops) | A component that renders a menu of options.

Reviewer: ```suggestion
AppointmentForm.Select | [AppointmentForm.SelectProps](#appointmentformselectprops) | A component that renders an options menu.
```

dateEditorComponent | ComponentType&lt;[AppointmentForm.DateEditorProps](#appointmentformdateeditorprops)&gt; | | A component that renders a date-time editor.
labelComponent | ComponentType&lt;[AppointmentForm.LabelProps](#appointmentformlabelprops)&gt; | | A component that renders a text label.
booleanEditorComponent | ComponentType&lt;[AppointmentForm.BooleanEditorProps](#appointmentformbooleaneditorprops)&gt; | | A component that renders an editor of Boolean values.
selectComponent | ComponentType&lt;[AppointmentForm.SelectProps](#appointmentformselectprops)&gt; | | A component that renders a menu of options.
radioGroupComponent | ComponentType&lt;[AppointmentForm.RadioGroupProps](#appointmentformradiogroupprops)&gt; | | A component that renders a radio group.
weeklyRecurrenceSelectorComponent | ComponentType&lt;[AppointmentForm.WeeklyRecurrenceSelectorProps](#appointmentformweeklyrecurrenceselectorprops)&gt; | | A component that renders a weekly recurrence selector.

## Interfaces

### Option

A menu option in the `selectComponent`.

Field | Type | Description
------|------|------------
id | string &#124; number | The option's identifier.

> text | string | The option's displayed text.

Reviewer: ```suggestion
text | string | The option's text.
```

isRecurrence | boolean | Specifies whether recurrence editors should be rendered.
children? | ReactNode | A React node used to render additional components to the layout.

### AppointmentForm.CommandLayoutProps

Properties passed to a component that renders a layout for command buttons.

Field | Type | Description
------|------|------------
commandButtonComponent | ComponentType&lt;[AppointmentForm.CommandButtonProps](#appointmentformcommandbuttonprops)&gt; | A component that renders a command button.
readOnly? | boolean | Specifies whether the appointment form is read-only.
fullSize | boolean | Specifies whether the command layout is full-size.
disableSaveButton? | boolean | Specifies whether to disable the Save button.
onCommitButtonClick | () => void | An event raised when the Commit button is clicked. The event handler should commit appointment changes.
onCancelButtonClick | () => void | An event raised when the Cancel button is clicked. The event handler should close the appointment form.
onDeleteButtonClick | () => void | An event raised when the Delete button is clicked. The event handler should delete an appointment.

> getMessage | (messageKey: string) => string | Retrieves a localization message using its key.

Reviewer: ```suggestion
getMessage | (messageKey: string) => string | Uses a localization message's key to retrieve the message.
```

formatDate | [FormatterFn](./scheduler.md#formatterfn) | A function that formats dates depending on the locale.
radioGroupComponent | ComponentType&lt;[AppointmentForm.RadioGroupProps](#appointmentformradiogroupprops)&gt; | A component that renders a radio group.
weeklyRecurrenceSelectorComponent | ComponentType&lt;[AppointmentForm.WeeklyRecurrenceSelectorProps](#appointmentformweeklyrecurrenceselectorprops)&gt; | A component that renders a weekly recurrence selector.
textEditorComponent | ComponentType&lt;[AppointmentForm.TextEditorProps](#appointmentformtexteditorprops)&gt; | A component that renders a text editor.
dateEditorComponent | ComponentType&lt;[AppointmentForm.DateEditorProps](#appointmentformdateeditorprops)&gt; | A component that renders a date-time editor.
selectComponent | ComponentType&lt;[AppointmentForm.SelectProps](#appointmentformselectprops)&gt; | A component that renders a menu of options.
labelComponent | ComponentType&lt;[AppointmentForm.LabelProps](#appointmentformlabelprops)&gt; | A component that renders a text label.
children? | ReactNode | A React node used to render additional components to the Basic Layout.

### AppointmentForm.BooleanEditorProps

> Properties passed to a component that renders an editor of Boolean values to the appointment form.

Reviewer: ```suggestion
Properties passed to a component that renders a Boolean value editor on the appointment form.
```

Field | Type | Description
------|------|------------
label? | string | The editor's text label.
readOnly? | boolean | Specifies whether the editor is read-only.
value? | boolean | The editor's value.
onValueChange | (nextValue: boolean) => void | Handles value changes.

### AppointmentForm.CommandButtonProps

Properties passed to a component that renders a command button to the appointment form.

Field | Type | Description
------|------|------------
id | `saveButton` &#124; `deleteButton` &#124; `cancelButton` | The command button's identifier.
onExecute | () => void | An event that initiates the command execution.
getMessage | (messageKey: string) => string | Retrieves a localization message using its key.

### AppointmentForm.DateEditorProps

> Properties passed to a component that renders a date-time editor to the appointment form.

Reviewer: to?

Field | Type | Description
------|------|------------
readOnly? | boolean | Specifies whether the date editor is read-only.
value? | string &#124; number &#124; Date | The editor's value.
onValueChange | (nextValue: Date) => void | Handles value changes.
locale? | string &#124; Array&lt;string&gt; | Specifies the locale as an IETF BCP 47 language tag or an array of such tags. The locale will be used to format date-time values.

### AppointmentForm.LabelProps

Properties passed to a component that renders a text label to the appointment form.

Field | Type | Description
------|------|------------
type? | `titleLabel` &#124; `ordinaryLabel` | The label's type.
text? | string | The label's text.

### AppointmentForm.RadioGroupProps

Properties passed to a component that renders a radio group to the appointment form.

Field | Type | Description
------|------|------------
appointmentData | [AppointmentModel](./scheduler.md#appointmentmodel) | The appointment's data.

> locale? | string &#124; Array&lt;string&gt; | Specifies the locale as an IETF BCP 47 language tag or an array of such tags. The locale will be used to format date-time values.

Reviewer: ```suggestion
locale? | string &#124; Array&lt;string&gt; | Specifies the locale as an IETF BCP 47 language tag or an array of such tags. The locale is used to format date-time values.
```

formatDate | [FormatterFn](./scheduler.md#formatterfn) | A function that formats dates depending on the locale.
onFieldChange | (nextFieldValue: { [fieldName: string]: any }) => void | An event raised when a field value in the appointment form is changed.
readOnly? | boolean | Specifies whether the radio group is read-only.
type? | `endRepeat` &#124; `monthlyRadioGroup` &#124; `yearlyRadioGroup` | The radio group's type.
getMessage | (messageKey: string) => string | Retrieves a localization message using its key.
textEditorComponent | ComponentType&lt;[AppointmentForm.TextEditorProps](#appointmentformtexteditorprops)&gt; | A component that renders a text editor.
dateEditorComponent | ComponentType&lt;[AppointmentForm.DateEditorProps](#appointmentformdateeditorprops)&gt; | A component that renders a date-time editor.
selectComponent | ComponentType&lt;[AppointmentForm.SelectProps](#appointmentformselectprops)&gt; | A component that renders a menu of options.
labelComponent | ComponentType&lt;[AppointmentForm.LabelProps](#appointmentformlabelprops)&gt; | A component that renders a text label.

### AppointmentForm.SelectProps

> Properties passed to a component that renders a menu of options to the appointment form.

Reviewer: Why 'renders' 'to'? Why 'to'?

onValueChange | (nextValue: string &#124; number) => void | Handles value changes.
availableOptions? | Array&lt;[Option](#option)&gt; | Specifies available menu options.
type? | `outlinedSelect` &#124; `filledSelect` | The menu's type.
readOnly | boolean | Specifies whether the menu is read-only.

### AppointmentForm.TextEditorProps

Properties passed to a component that renders a text editor to the appointment form.

Field | Type | Description
------|------|------------
placeholder | string | A placeholder displayed inside the text field.
type | `titleTextEditor` &#124; `noteTextEditor` &#124; `ordinaryTextEditor` &#124; `numberEditor` | The text editor's type.
readOnly | boolean | Specifies whether the text editor is read-only.
value | string &#124; number | The editor's value.
onValueChange | (nextValue: string) => void | Handles value changes.

### AppointmentForm.WeeklyRecurrenceSelectorProps

Properties passed to a component that renders a weekly recurrence selector to the appointment form.

Field | Type | Description
------|------|------------
rRule | string | Specifies the recurrence rule.
readOnly | boolean | Specifies whether the weekly recurrence selector is read-only.

> formatDate | [FormatterFn](./scheduler.md#formatterfn) | A function that formats dates depending on the locale.

Reviewer 2: It's not that there are dates that depend on the locale and dates that don't. This function uses different date formatting depending on which locale is currently selected.

> formatDate | [FormatterFn](./scheduler.md#formatterfn) | A function that formats dates depending on the locale.

Reviewer: ```suggestion
formatDate | [FormatterFn](./scheduler.md#formatterfn) | A function that formats dates based on the locale.
```
